### PR TITLE
Get tests to build as part of pipeline

### PR DESCRIPTION
**Pull request by rchiodo**
Pipeline wasn't building test project (so no tests to run when creating test pipeline).

### CODE REVIEW

**Comment by rchiodo** [comment]
This was getting too big. Going to submit this now and continue working on fixing more tests.

File: Python/Tests/Utilities.Python.Analysis/PythonPaths.cs [hunk 1]
```diff
@@ -215,21 +241,32 @@ public static class PythonVersionExtensions {
...
         /// Creates a Python virtual environment in specified directory.
         /// </summary>
-        public static void CreateVirtualEnv(this PythonVersion pyVersion, string envPath) {
-            var virtualEnvModule = (pyVersion.Version < PythonLanguageVersion.V30) ? "virtualenv" : "venv";
-            using (var p = ProcessOutput.RunHiddenAndCapture(pyVersion.InterpreterPath, "-m", virtualEnvModule, envPath)) {
-                Console.WriteLine(p.Arguments);
-                Assert.IsTrue(p.Wait(TimeSpan.FromMinutes(3)));
-                Console.WriteLine(string.Join(Environment.NewLine, p.StandardOutputLines.Concat(p.StandardErrorLines)));
-                Assert.AreEqual(0, p.ExitCode);
+        public static string CreateVirtualEnv(this PythonVersion pyVersion, VirtualEnvName envName, string rootDirectory = null) {
```
Comment (rchiodo) [inline Python/Tests/Utilities.Python.Analysis/PythonPaths.cs:250]: This change here is to speed up tests. Creating VirtualEnvs takes like 10 seconds each when tests can share the same virtual env.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -162,112 +171,113 @@ steps:
...
+- task: PublishBuildArtifacts@1
+  condition: notin(variables['Build.Reason'], 'PullRequest')
+  displayName: 'Upload staging zip to build artifacts'
+  inputs:
+    PathtoPublish: '$(Build.ArtifactStagingDirectory)/PTVSBuildArchive.zip'
+    ArtifactName: 'PTVS Staging Output'
+
+# Publish test data archive (so we don't have to pull down everything to run tests)
+- task: PublishBuildArtifacts@1
+  displayName: 'Upload test data zip to build artifacts'
```
Comment (rchiodo) [inline azure-pipelines.yml:280]: This testData artifact is used by the build pipeline here:
https://devdiv.visualstudio.com/DevDiv/_taskgroup/a7b18468-aab5-4be7-b718-02f85b0828c4

Comment (rchiodo) [reply azure-pipelines.yml:280]: I should probably add that as a comment in here so people know how it's used.

File: Python/Tests/Utilities.Python.Analysis/PythonPaths.cs [hunk 3]
```diff
@@ -120,14 +127,32 @@ public class PythonPaths {
...
                 if (Python36_x64 != null) yield return Python36_x64;
                 if (Python37_x64 != null) yield return Python37_x64;
-                if (Python38_x64 != null) yield return Python36_x64;
-                if (Python39_x64 != null) yield return Python37_x64;
+                if (Python38_x64 != null) yield return Python38_x64;
+                if (Python39_x64 != null) yield return Python39_x64;
+            }
+        }
+
+        public static PythonVersion LatestVersion {
```
Comment (rchiodo) [inline Python/Tests/Utilities.Python.Analysis/PythonPaths.cs:141]: This property makes the tests runnable with whatever the latest Python is on your box.

File: Python/Tests/Utilities.Python.Analysis/PythonPaths.cs (hunk 1 above)
Comment (bschnurr) [reply Python/Tests/Utilities.Python.Analysis/PythonPaths.cs:250]: I thought I had a change similar to this in testing.

Comment (bschnurr) [reply Python/Tests/Utilities.Python.Analysis/PythonPaths.cs:250]: this was the change. https://github.com/microsoft/PTVS/pull/5505

Comment (rchiodo) [reply Python/Tests/Utilities.Python.Analysis/PythonPaths.cs:248]: I believe that change would cache the source and results folders. Virtual Envs are being created in the temp folder.

Comment (rchiodo) [reply Python/Tests/Utilities.Python.Analysis/PythonPaths.cs:248]: Well at least some of them are.

**Comment by rchiodo** [comment]
@bschnurr did you have any more comments?

**Comment by rchiodo** [comment]
Still working on getting all the tests to pass, but wanted to submit this first.

**Pull request merged**